websites/F/Fandom/presence.ts

```diff
@@ -1,356 +1,373 @@
-      presenceData.details = "Viewing a topic";
-      presenceData.state = document.querySelector(
-        ".topic-header__title"
-      ).firstElementChild.innerHTML;
-    } else if (currentPath[0] === "video") {
-      presenceData.details = "Watching a video";
-      updateCallback.function = (): void => {
-        presenceData.state = document.querySelector(
-          ".video-page-featured-player__title"
-        ).textContent;
-        try {
-          if (
-            document
-              .querySelector(".jw-icon-playback")
-              .getAttribute("aria-label") === "Pause"
-          ) {
-            const video: HTMLVideoElement = document.querySelector(".jw-video"),
-              timestamps = presence.getTimestampsfromMedia(video);
-            presenceData.startTimestamp = timestamps[0];
-            presenceData.endTimestamp = timestamps[1];
-          } else {
-            delete presenceData.startTimestamp;
-            delete presenceData.endTimestamp;
-          }
-        } catch (e) {
-          delete presenceData.startTimestamp;
-          delete presenceData.endTimestamp;
+        if (currentPath[0] === "") {
+          presenceData.details = "On the index page";
+        } else if (currentPath[0] === "signin") {
+          presenceData.details = "Signing in";
+        } else if (currentPath[0] === "register") {
+          presenceData.details = "Registering an account";
+        } else if (currentPath[0] === "articles") {
+          presenceData.details = "Reading an article";
+          presenceData.state = document.querySelector(
+            ".article-header__title"
+          ).textContent;
+        } else if (currentPath[0] === "topics") {
+          presenceData.details = "Viewing a topic";
+          presenceData.state = document.querySelector(
+            ".topic-header__title"
+          ).firstElementChild.innerHTML;
+        } else if (currentPath[0] === "video") {
+          presenceData.details = "Watching a video";
+          updateCallback.function = (): void => {
+            presenceData.state = document.querySelector(
+              ".video-page-featured-player__title"
+            ).textContent;
+            try {
+              if (
+                document
+                  .querySelector(".jw-icon-playback")
+                  .getAttribute("aria-label") === "Pause"
+              ) {
+                const video: HTMLVideoElement = document.querySelector(
+                    ".jw-video"
+                  ),
+                  timestamps = presence.getTimestampsfromMedia(video);
+                presenceData.startTimestamp = timestamps[0];
+                presenceData.endTimestamp = timestamps[1];
+              } else {
+                delete presenceData.startTimestamp;
+                delete presenceData.endTimestamp;
+              }
+            } catch (e) {
+              delete presenceData.startTimestamp;
+              delete presenceData.endTimestamp;
+            }
```

Comment:
```suggestion
                presenceData.endTimestamp = timestamps[1];
              } else
                delete presenceData.endTimestamp;
            } catch (e)
              delete presenceData.endTimestamp;
```
i think I did it good

Reply:
Not sure if that's okay. I'm following [the docs](https://docs.premid.app/en/dev/presence/class#gettimestampsnumber-number).

Reply:
idk who did it but if you have endTimestamp you don't need startTimestamp.

Reply:
![image](https://user-images.githubusercontent.com/9348108/114399294-0d4b2300-9ba1-11eb-9be4-67652e36ab5e.png)
basically, if you have endTimestamp it shows remaining no matter what

Reply:
But, here's a thing. I'm not sure if there are hidden intricacies related to the RPC. I'm not sure if Discord actually uses only either one of them, but I just wanted to play safe like what I've done since the inception of this presence.

I would like to keep the `startTimestamp` as this seems the standard implementation in a lot of presences, [including YouTube](https://github.com/PreMiD/Presences/blob/master/websites/Y/YouTube/presence.ts#L240-L241) and this seemed next to unnecessary, but if is really, really a necessary change, please let me know. I would still fix the next error.

Reply:
Looks it works like that:
 - If ``startTimestamp`` is in RPC object: Discord shows elapsed.
 - If ``endTimestamp`` is in RPC object: Discord shows remaining.
 - If ``startTimestamp`` and ``endTimestamp`` is in RPC object: Discord shows remaining. 
 That's how it works. I know that other presences use it, but literally I'm not going to tell creator aka Bas950 to delete it.

Reply:
waiting...

Reply:
Hello, it's been a while since I touched GH for a while. I was both shifted to my other smaller projects, and preparing for the college entry exams. Kind of surprised that this doesn't get merged after a while.

I forgot to tell you, but I have asked Bas950 about this, and he told me to "keep it like the rest of the presences." This is conflicting, so I hope you can resolve it with him. But for now, I will leave this as-is. 

I just hope that this could be merged in a quick fashion due to a critical bug that I fixed a while ago, which kind of overshadowed by this issue.

Reply:
Bas agreed with me rn after some discussion so you are feel free to delete this line.

Reply:
Alright, thank god we got that sorted out on the back. I have did the changes that you requested, but unfortunately I need to add a single line since it seems that you assume `startTimestamp` is empty.

Two things that I wanted to tell you is that (a) if something goes wrong, you will be at the fault, and (b) I hope this kind of debate won't be repeated again in the future.

In the end, I apologize for the inconvenience.

Reply:
Everything is ok, just test if that works on this specific subpage.